android/java/org/chromium/chrome/browser/crypto_wallet/util/ImageLoader.java

```diff
@@ -5,47 +5,363 @@
+import org.chromium.content_public.browser.WebContents;
+import org.chromium.url.GURL;
 
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
 
 public class ImageLoader {
     private static final List<String> ANIMATED_LIST = Arrays.asList(".gif");
+    private static final String UNUSED_CLIENT_NAME = "unused";
+    private static final String BASE64_ENCODING_PATTERN =
+            "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$";
+
+    /**
+     * Downloads an image from a given URL, including support for GIF and SVG image types.
+     * @param url URL of the image to download.
+     * @param context Android context used by Glide for applying transformations.
+     * @param isCircular When {@code true}, a circular transformation will be applied.
+     * @param imageView ImageView where the downloaded image will be set.
+     * @param callback Callback used to notify if the image has been set correctly. It can be {@code
+     *         null}.
+     */
+    public static void downloadImage(String url, final Context context, final boolean isCircular,
+            final ImageView imageView, final Callback callback) {
+        if (!isValidImgUrl(url)) {
+            if (callback != null) callback.onLoadFailed();
+            return;
+        }
+
+        Profile profile = Utils.getProfile(false);
+        if (isSvg(url)) {
+            final String validUrl;
+            if (URLUtil.isDataUrl(url)) {
+                if (isBase64Encoded(url)) {
+                    String decodedUrl = decodeBase64SvgUrl(url);
+                    validUrl = sanitizeSvg(decodedUrl);
+                } else if (isUtf8(url)) {
+                    validUrl = sanitizeSvg(url);
+                } else {
+                    // Unsupported URL type.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+                if (TextUtils.isEmpty(validUrl)) {
+                    // This may happen for invalid, or corrupted URLs.
+                    if (callback != null) callback.onLoadFailed();
+                    return;
+                }
+            } else {
+                validUrl = url;
+            }
+
+            WebContentsFactory webContentsFactory = new WebContentsFactory();
+            WebContents webContents =
+                    webContentsFactory.createWebContentsWithWarmRenderer(profile, false);
```

Comment:
Also I don't understand why upstream didn't make `createWebContentsWithWarmRenderer` as a static member as all other members are static there. Maybe let's use a static call to avoid an extra object creation? 
`WebContents webContents = WebContentsFactory.createWebContents(profile, true, true);`

Reply:
<img width="998" alt="WebContentsFactory_java_-_Chromium_Code_Search" src="https://user-images.githubusercontent.com/3308503/224309750-af0aee12-e7f3-4630-80c0-c7baeffce985.png">
The static method cannot be referenced, it's marked as private.

Reply:
ah, ok, I didn't see that, yeah, so we have to create the factory in any case, still not sure why, but ...:).

Reply:
🤷‍♂️